File: ckgame-java/src/main/java/com/ckgame/events/StorylineStage.java
package com.ckgame.events;

import java.util.List;

/**
 * 剧情线阶段。
 */
public final class StorylineStage {
    public final int stageId;
    public final String title;
    public final String description;
    public final List<Integer> eventIds;
    public final Integer nextStage;
    public final String requiresCondition;

    public StorylineStage(int stageId, String title, String description,
                          List<Integer> eventIds, Integer nextStage, String requiresCondition) {
        this.stageId = stageId;
        this.title = title;
        this.description = description;
        this.eventIds = eventIds;
        this.nextStage = nextStage;
        this.requiresCondition = requiresCondition;
    }

    public StorylineStage(int stageId, String title, String description, List<Integer> eventIds) {
        this(stageId, title, description, eventIds, null, null);
    }
}

File: ckgame-ts/src/events/StorylineStage.ts
/**
 * A storyline stage.
 */
export class StorylineStage {
  public readonly stageId: number;
  public readonly title: string;
  public readonly description: string;
  public readonly eventIds: number[];
  public readonly nextStage: number | null;
  public readonly requiresCondition: string | null;

  constructor(
    stageId: number,
    title: string,
    description: string,
    eventIds: number[],
    nextStage?: number | null,
    requiresCondition?: string | null,
  ) {
    this.stageId = stageId;
    this.title = title;
    this.description = description;
    this.eventIds = eventIds;
    this.nextStage = nextStage ?? null;
    this.requiresCondition = requiresCondition ?? null;
  }
}
